ascmhl/generator.py

```diff
@@ -41,6 +41,78 @@ def __init__(self, history: MHLHistory, ignore_spec: MHLIgnoreSpec = MHLIgnoreSp
         self.new_hash_lists = defaultdict(MHLHashList)
         self.ignore_spec = ignore_spec
 
+    def append_multiple_format_file_hashes(
+        self, file_path, file_size, hash_lookup: Dict[str, str], file_modification_date, action=None, hash_date=None
+    ) -> bool:
+        """
+        Adds file hashes to the history
+        :param file_path: a string value representing the path to a file
+        :param file_size: size of the file path in bytes
+        :param hash_lookup: a dictionary of hash values keyed by the respective hash format
+        :param file_modification_date: date the file was last modified
+        :param action: a predetermined action for the entry.  defaults to none
+        :param hash_date: date the hashes were generated
+        :return a bool indicating if the hashes were successfully appended.  returns false if any failures occur
+        """
+        relative_path = self.root_history.get_relative_file_path(file_path)
+        # TODO: handle if path is outside of history root path
+        # Keep track of the number of failures
+        failures = 0
+        history, history_relative_path = self.root_history.find_history_for_path(relative_path)
+        # for collections we cannot create a valid relative path (we are in the "wrong" history), but in that case
+        # the file_path is inputted already as the relative path (a bit of implicit functionality here)
+        if history_relative_path == None:
+            history_relative_path = file_path
+
+        # check if there is an existing hash in the other generations and verify
+        original_hash_entry = history.find_original_hash_entry_for_path(history_relative_path)
+
+        hash_entries = [MHLHashEntry]
+        # TODO: sort the format keys into a standard order for consistent output
```

Comment:
I think this TODO can be eliminated

Reply:
It looks to me as if the method `append_multiple_format_file_hashes` is currently not used. Did you add that temporarily?

Reply:
I believe I created this function as a way to streamline appending of file hash values to the session, however the current implementation did not lend itself to utilizing this as in certain cases it is only desirable to append a subsection of the hashes generated.  There may be a use case for this in the future, but the way things are currently implemented there is not without some refactor work.  I'll leave it to you whether to keep this function or remove it